main/src/kotlinx/team/infra/InfraExtension.kt

```diff
@@ -26,12 +26,18 @@ open class InfraExtension(val project: Project) {
         teamcityHandler?.invoke(teamcity)
     }
 
+    @Deprecated("No longer supported. Use binary-compatibility-validator instead.")
+    @Suppress("DEPRECATION")
     val apiCheck = project.objects.newInstance<ApiCheckConfiguration>()
+    @Suppress("DEPRECATION")
     private var apiCheckHandler: ((ApiCheckConfiguration) -> Unit)? = null
-    internal fun afterApiCheck(handler: (ApiCheckConfiguration) -> Unit) {
+    @Suppress("DEPRECATION")
+internal fun afterApiCheck(handler: (ApiCheckConfiguration) -> Unit) {
         apiCheckHandler = handler
     }
 
+    @Deprecated("No longer supported. Use binary-compatibility-validator instead.")
+    @Suppress("DEPRECATION")
     fun apiCheck(configure: Action<ApiCheckConfiguration>) {
```

Comment:
Perhaps we could drop this functionality immediately if we can make sure nobody uses it.

Reply:
I am not aware of any library that uses it.

Reply:
But I don't mind having it deprecated for some time.